ompi/communicator/comm_cid.c

```diff
@@ -963,6 +1029,64 @@ int ompi_comm_activate (ompi_communicator_t **newcomm, ompi_communicator_t *comm
     return rc;
 }
 
+int ompi_comm_get_remote_cid (ompi_communicator_t *comm, int dest, uint32_t *remote_cid)
+{
+    ompi_proc_t *ompi_proc;
+    pmix_proc_t  pmix_proc;
+    pmix_info_t tinfo[2];
+    pmix_value_t *val = NULL;
+    ompi_comm_extended_cid_t excid;
+    int rc = OMPI_SUCCESS;
+    size_t remote_cid64;
+
+    assert(NULL != remote_cid);
+
+    if (OMPI_COMM_IS_GLOBAL_INDEX(comm)) {
+        *remote_cid = comm->c_index;
+    } else {
+        ompi_proc = ompi_comm_peer_lookup(comm, dest);
+        OPAL_PMIX_CONVERT_NAME(&pmix_proc, &ompi_proc->super.proc_name);
+
+        PMIx_Info_construct(&tinfo[0]);
+        PMIX_INFO_LOAD(&tinfo[0], PMIX_TIMEOUT, &ompi_pmix_connect_timeout, PMIX_UINT32);
+
+        excid =  ompi_comm_get_extended_cid (comm);
+
+        PMIX_INFO_CONSTRUCT(&tinfo[1]);
+        PMIX_INFO_LOAD(&tinfo[1], PMIX_GROUP_CONTEXT_ID, &excid.cid_base, PMIX_SIZE);
+        PMIX_INFO_SET_QUALIFIER(&tinfo[1]);
+        if (PMIX_SUCCESS != (rc = PMIx_Get(&pmix_proc, PMIX_GROUP_LOCAL_CID, tinfo, 2, &val))) {
+                OPAL_OUTPUT_VERBOSE((10, ompi_comm_output, "PMIx_Get failed for PMIX_GROUP_LOCAL_CID cid_base %ld %s", excid.cid_base, PMIx_Error_string(rc)));
```

Comment:
```suggestion
            OPAL_OUTPUT_VERBOSE((10, ompi_comm_output, "PMIx_Get failed for PMIX_GROUP_LOCAL_CID cid_base %ld %s", excid.cid_base, PMIx_Error_string(rc)));
            rc = opal_pmix_convert_status(rc);
            goto done;
```

, and please remove `if (PMIX_SUCCESS == rc) {`.